### PR TITLE
[PLAT-3907] Add reset_app method with 0.1 second timeout by default

### PR DESCRIPTION
**Pull request by Cawllec**
## Goal
Reduces the chance that we get duplicate requests from the test fixture closing too quickly.  While this doesn't deterministically deal with the issue, it should provide some more resilience by allowing extra time for HTTP requests to resolve.

In order to use this method, `$driver.reset` calls in the `After` blocks that use AppAutomateDriver should be changed to `$driver.reset_app`

## Design
The 0.1 second default timeout is intended to allow HTTP requests in progress to resolve before the app is reset, while also being small enough that it shouldn't have a large impact on the overall runtime of tests.  This value is variable, so can be adjusted on a per notifier basis.

## Tests
The intended use has been added to the browserstack test fixture.

### CODE REVIEW

File: lib/features/support/app_automate_driver.rb [hunk 1]
```diff
@@ -122,6 +122,14 @@ def clear_and_send_keys_to_element(element_id, text)
     element.send_keys(text)
   end
 
+  # Reset the currently running application after a given timeout
+  #
+  # @param timeout [Number] the amount of time in seconds to wait before resetting
+  def reset_app(timeout=0.1)
```
[the remark this answers is not in the document]
Comment (twometresteve) [reply lib/features/support/app_automate_driver.rb:128]: As it turns out, I'm now using a similar technique to wrap an exception handler and Driver `restart` around `reset`... so probably best that I let you get this in first and I'll rebase my changes onto it.

[the remark this answers is not in the document]
Comment (Cawllec) [reply lib/features/support/app_automate_driver.rb:128]: I guess my thought is that stopping the use of the `reset` method by overwriting it completely will stop us controlling or adjusting the timings on a per-scenario basis.  I wouldn't object to configuring the timeout as a parameter but allowing it to be overwritten on the method calls (thus allowing us to "zero" it if necessary)

[the remark this answers is not in the document]
Comment (twometresteve) [reply lib/features/support/app_automate_driver.rb:128]: Yeah, I can see it's more flexible tis way and it also provides the class with a slightly higher level of abstraction than using the Appium driver directly.

**Review by twometresteve — APPROVED** [review]
Approved - if you could just add a CHANGELOG entry.

**Review by twometresteve — APPROVED** [review]
LGTM

**Pull request merged**